### PR TITLE
Feature/dialog component interface

### PR DESCRIPTION
**Pull request by adnan1naeem**
## What I did
Modal Interface
## How to test
no
### Is this testable with jest or storyshots?
no
### Are any of the existing tests are failing ?
no
### Does this need an update to the documentation?
no
### Does it have breaking changes ?
no
If your answer is yes to any of these, please make sure to include it in your PR.

@bjavaid @amnajunaid please review the changes and merge them.

### CODE REVIEW

File: src/components/Modal.ts [hunk 1]
```diff
@@ -25,7 +26,7 @@ export interface ModalProperties {
 	 * It is for iOS only
 	 */
 
-	onDismiss: () => void;
+	onDismiss?: () => void;
```
Comment (amnajunaid) [inline src/components/Modal.ts]: @adnan1naeem  This should be mandatory. Please discuss

Comment (adnan1naeem) [reply src/components/Modal.ts]: we have discussed regarding this

**Pull request merged**